graalvm/build.gradle

```diff
@@ -96,13 +96,36 @@ task getGraalBuildFolder(type: Exec) {
     }
 }
 
+/**
+ * SubstrateVM can add required native libraries during compilation based on parameters to
+ * native-image. For example, the inclusion of --enable-all-security-services results in native-image
+ * automatically including libstdc++.a during linking. We link against the SGX trusted runtime to
+ * satisfy the linker so to prevent multiply derived symbols we need to provide placeholder, empty
+ * implementations of any automatically linked library. This function populates the placeholderlibs/
+ * directory with these files and NativeImage.kt adds this directory to the link path
+ */
+task generatePlaceholderLibraries(type: Exec) {
+    dependsOn getGraalBuildFolder
+
+    // Task details
+    group = 'other'
+    description = 'Creates and populates the placeholderlibs directory inside the Graal folder.'
+
+    doFirst {
+        def placeholderDir = getGraalBuildFolder.graalVMDir + "/placeholderlibs/"
```

Comment:
I'm not a big fan of this global `graalVMDir` spread across different tasks. There's an implicit dependency created between the tasks that use this dir, which is not obvious. Ideally we should be using gradle tasks inputs and outputs to create the dependency explicitly, and the output `graalVMDir` only ever needs to be defined once. Not sure how easy this is, and probably not for this PR.